static/app/views/monitors/components/overviewTimeline/checkInTimeline.tsx

```diff
@@ -27,12 +27,12 @@ function getBucketedCheckInsPosition(
 }
 
 export function CheckInTimeline(props: Props) {
-  const {bucketedData, start, end, timeWindow, width} = props;
+  const {bucketedData, start, end, timeWindow, width, environments} = props;
 
   const elapsedMs = end.getTime() - start.getTime();
   const msPerPixel = elapsedMs / width;
 
-  const jobTicks = mergeBuckets(bucketedData);
+  const jobTicks = mergeBuckets(bucketedData, environments);
```

Comment:
Went back and forth on where the filtering should be, but ended up deciding that we can just pass in a set of environments to the timeline and have it handle filtering since it is already processing the data of each bucket for merging.